metricbeat/module/prometheus/remote_write/remote_write_test.go

```diff
@@ -78,3 +78,188 @@ func TestGenerateEventsCounter(t *testing.T) {
+					Value:  500,
+				},
+			},
+			expected: map[string]int{
+				`{"handler":"/home","le":"+Inf"}`: 1,
+				`{"handler":"/home"}`:             2,
+				`{"handler":"/home","le":"0.1"}`:  1,
+				`{"handler":"/home","le":"0.5"}`:  1,
+			},
+		},
+		{
+			name: "Mix of counter, gauge, and histogram",
+			samples: model.Samples{
+				&model.Sample{
+					Metric: model.Metric{"__name__": "http_requests_total", "method": "GET", "status": "200"},
+					Value:  100,
+				},
+				&model.Sample{
+					Metric: model.Metric{"__name__": "cpu_usage", "core": "0"},
+					Value:  45.5,
+				},
+				&model.Sample{
+					Metric: model.Metric{"__name__": "request_duration_seconds_bucket", "le": "0.1"},
+					Value:  30,
+				},
+				&model.Sample{
+					Metric: model.Metric{"__name__": "request_duration_seconds_bucket", "le": "0.5"},
+					Value:  50,
+				},
+				&model.Sample{
+					Metric: model.Metric{"__name__": "request_duration_seconds_sum"},
+					Value:  75.5,
+				},
+				&model.Sample{
+					Metric: model.Metric{"__name__": "request_duration_seconds_count"},
+					Value:  60,
+				},
+			},
+			expected: map[string]int{
+				`{"le":"0.1"}`:                    1,
+				`{"le":"0.5"}`:                    1,
+				`{"method":"GET","status":"200"}`: 1,
+				`{"core":"0"}`:                    1,
+				`{}`:                              2,
+			},
+		},
+		{
+			name: "Duplicate labels and distinct labels",
+			samples: model.Samples{
+				&model.Sample{
+					Metric: model.Metric{"__name__": "api_calls", "endpoint": "/users", "method": "GET"},
+					Value:  50,
+				},
+				&model.Sample{
+					Metric: model.Metric{"__name__": "api_calls", "endpoint": "/users", "method": "POST"},
+					Value:  30,
+				},
+				&model.Sample{
+					Metric: model.Metric{"__name__": "api_calls", "endpoint": "/products", "method": "GET"},
+					Value:  40,
+				},
+				&model.Sample{
+					Metric: model.Metric{"__name__": "system_load", "host": "server1"},
+					Value:  1.5,
+				},
+				&model.Sample{
+					Metric: model.Metric{"__name__": "system_load", "host": "server2"},
+					Value:  2.0,
+				},
+			},
+			expected: map[string]int{
+				`{"endpoint":"/users","method":"GET"}`:    1,
+				`{"endpoint":"/users","method":"POST"}`:   1,
+				`{"endpoint":"/products","method":"GET"}`: 1,
+				`{"host":"server1"}`:                      1,
+				`{"host":"server2"}`:                      1,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			generator := RemoteWriteEventGenerator{
+				metricsCount: true,
+			}
+
+			events := generator.GenerateEvents(tt.samples)
+
+			assert.Equal(t, len(tt.expected), len(events), "Number of generated events should match expected")
+
+			for _, event := range events {
+				count, ok := event.RootFields["metrics_count"]
+				assert.True(t, ok, "metrics_count should be present")
+
+				labels, ok := event.ModuleFields["labels"].(mapstr.M)
```

Comment:
I noticed that we are using  `assert.True(t, ok, "labels should be present")` in the x-pack counterpart. Any reason that we shouldn't be doing the same thing here?

Reply:
No. Both should be similar. I'll make them same. Changing in x-pack.

Reply:
Change: https://github.com/elastic/beats/pull/40411/commits/274f104f49e177072726ef20410442be5f43be78